drivers/afe/ad4110/ad4110.c

```diff
@@ -196,118 +273,130 @@ int32_t ad4110_set_op_mode(struct ad4110_dev *dev, enum ad4110_op_mode mode)
-						    A4110_AFE,
-						    AD4110_REG_AFE_CNTRL2,
-						    AD4110_ENABLE,
-						    AD4110_REG_AFE_CNTRL2_EN_FLD_PWR_MSK);
+		ret = ad4110_spi_int_reg_write_msk(dev,
+						   A4110_AFE,
+						   AD4110_REG_AFE_CNTRL2,
+						   AD4110_ENABLE,
+						   AD4110_REG_AFE_CNTRL2_EN_FLD_PWR_MSK);
 
 		break;
-
 	case AD4110_RTD_4W_MODE:
 		// clear IMODE bit
 		ret = ad4110_spi_int_reg_write_msk(dev,
 						   A4110_AFE,
 						   AD4110_REG_AFE_CNTRL2,
 						   AD4110_DISABLE,
 						   AD4110_REG_AFE_CNTRL2_IMODE_MSK);
+		if (ret)
+			return ret;
+
 		// clear RTD_3W4W bit
-		ret |= ad4110_spi_int_reg_write_msk(dev,
-						    A4110_AFE,
-						    AD4110_REG_PGA_RTD_CTRL,
-						    AD4110_DISABLE,
-						    AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
+		ret = ad4110_spi_int_reg_write_msk(dev,
+						   A4110_AFE,
+						   AD4110_REG_PGA_RTD_CTRL,
+						   AD4110_DISABLE,
+						   AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
 		break;
-
 	case AD4110_RTD_3W_MODE:
 		// clear IMODE bit
 		ret = ad4110_spi_int_reg_write_msk(dev,
 						   A4110_AFE,
 						   AD4110_REG_AFE_CNTRL2,
 						   AD4110_DISABLE,
 						   AD4110_REG_AFE_CNTRL2_IMODE_MSK);
+		if (ret)
+			return ret;
+
 		// set RTD_3W4W bit
-		ret |= ad4110_spi_int_reg_write_msk(dev,
-						    A4110_AFE,
-						    AD4110_REG_PGA_RTD_CTRL,
-						    AD4110_ENABLE,
-						    AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
+		ret = ad4110_spi_int_reg_write_msk(dev,
+						   A4110_AFE,
+						   AD4110_REG_PGA_RTD_CTRL,
+						   AD4110_ENABLE,
+						   AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
 		break;
-
 	case AD4110_RTD_2W_MODE:
 		// clear IMODE bit
 		ret = ad4110_spi_int_reg_write_msk(dev,
 						   A4110_AFE,
 						   AD4110_REG_AFE_CNTRL2,
 						   AD4110_DISABLE,
 						   AD4110_REG_AFE_CNTRL2_IMODE_MSK);
+		if (ret)
+			return ret;
+
 		// set RTD_3W4W bit
-		ret |= ad4110_spi_int_reg_write_msk(dev,
-						    A4110_AFE,
-						    AD4110_REG_PGA_RTD_CTRL,
-						    AD4110_ENABLE,
-						    AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
+		ret = ad4110_spi_int_reg_write_msk(dev,
+						   A4110_AFE,
+						   AD4110_REG_PGA_RTD_CTRL,
+						   AD4110_ENABLE,
+						   AD4110_REG_PGA_RTD_CTRL_23W_EN_MSK);
+		if (ret)
+			return ret;
+
 		// enable VBIAS
-		ret |=
-			ad4110_spi_int_reg_write_msk(dev,
-						     A4110_AFE,
-						     AD4110_REG_AFE_CNTRL2,
-						     AD4110_AFE_VBIAS(AD4110_AFE_VBIAS_ON),
-						     AD4110_AFE_VBIAS(AD4110_AFE_VBIAS_OFF));
+		ret = ad4110_spi_int_reg_write_msk(dev,
+						   A4110_AFE,
+						   AD4110_REG_AFE_CNTRL2,
+						   AD4110_AFE_VBIAS(AD4110_AFE_VBIAS_ON),
+						   AD4110_AFE_VBIAS(AD4110_AFE_VBIAS_OFF));
+		if (ret)
+			return ret;
 
 		// enable the 100 �A pull-down current source on AIN(-)
-		ret |= ad4110_spi_int_reg_write_msk(dev,
-						    A4110_AFE,
-						    AD4110_REG_AFE_CNTRL2,
-						    AD4110_REG_AFE_CNTRL2_AINN_DN100,
-						    AD4110_REG_AFE_CNTRL2_AINN_DN100);
+		ret = ad4110_spi_int_reg_write_msk(dev,
```

Comment:
return directly. after implementing all the above returns, you can also remove `return ret;` at the end of the function